components/match_ticker/commons/match_ticker_display_components_new.lua

```diff
@@ -0,0 +1,366 @@
+---@operator call(table): NewMatchTickerMatch
+---@field root Html
+---@field hideTournament boolean
+---@field onlyHighlightOnValue string?
+---@field match table
+local Details = Class.new(
+	function(self, args)
+		assert(args.match, 'No Match passed to MatchTickerDetails class')
+		self.root = mw.html.create('div'):addClass('match-details')
+		self.hideTournament = args.hideTournament
+		self.onlyHighlightOnValue = args.onlyHighlightOnValue
+		self.match = args.match
+	end
+)
+
+---@return Html
+function Details:create()
+	local highlightCondition = HighlightConditions.match or HighlightConditions.tournament
+	if highlightCondition(self.match, {onlyHighlightOnValue = self.onlyHighlightOnValue}) then
+		self.root:addClass(HIGHLIGHT_CLASS)
+	end
+
+	return self.root
+		:node(self:streams())
+		:node(self:tournament())
+		:node(self:countdown())
+end
+
+---It will display both countdown and date of the match so the user can select which one to show
+---@return Html
+function Details:countdown()
+	local match = self.match
+
+	local dateString
+	if Logic.readBool(match.dateexact) then
+		local timestamp = DateExt.readTimestamp(match.date) + (Timezone.getOffset(match.extradata.timezoneid) or 0)
+		dateString = DateExt.formatTimestamp('F j, Y - H:i', timestamp) .. ' '
+				.. (Timezone.getTimezoneString(match.extradata.timezoneid) or (Timezone.getTimezoneString('UTC')))
+	else
+		dateString = mw.getContentLanguage():formatDate('F j, Y', match.date) .. (Timezone.getTimezoneString('UTC'))
+	end
+
+	local countdownArgs = {
+		date = dateString,
+		finished = match.finished
+	}
+
+	local countdownDisplay = mw.html.create('span')
+		:addClass('match-countdown')
+		:node(Countdown._create(countdownArgs))
+
+	return countdownDisplay
+end
+
+---@return Html?
+function Details:streams()
+	local match = self.match
+	local links = mw.html.create('div')
+		:addClass('match-streams')
+
+	if Table.isNotEmpty(match.stream) then
+		local streams = {}
+
+		-- Standardize the stream data to always use the platform as key (because of the new format ex: twitch_en_2)
+		for rawHost, stream in pairs(match.stream) do
+			local streamParts = mw.text.split(rawHost, '_', true)
+			if #streamParts == 3 then
+				local key = StreamLinks.StreamKey(rawHost)
+				streams[key.platform] = stream
+			else
+				streams[rawHost] = stream
+			end
+		end
+
+		local streamLinks = ''
+
+		-- Iterate over the streams and create the different links
+		for platformName, targetStream in pairs(streams) do
+			local streamLink = mw.ext.StreamPage.resolve_stream(platformName, targetStream)
+
+			if streamLink then
+				-- Default values
+				local url = 'Special:Stream/' .. platformName .. '/' .. streamLink
+				local icon = '<i class="lp-icon lp-icon-21 lp-' .. platformName .. '"></i>'
+
+				-- TL.net specific
+				if platformName == 'stream' then
+					url = 'https://tl.net/video/streams/' .. streamLink
+				end
+
+				streamLinks = streamLinks .. '[[' .. url .. '|' .. icon .. ']]'
```

Comment:
could use module:page here (can be ignored if we first do the countdown update)

Reply:
The space between the icons is dealt with in the CSS
For the stream yes good catch, can update it depending on when we want to do the countdown update